### ISSUE TITLE
Universal sentence encoder training

### ISSUE THREAD

**Issue by saikrishna9494** [tensorflow/hub#36]
I want to use the universal sentence encoder for clustering. But the problem is I am having a different domain of data. So Is it possible to train my data to generate the sentence vectors?

**Comment by andresusanopinto** [comment]
Is your questions whether you can fine-tune the module to your data?
- In general the answer is a fuzzy "yes". Note that not all parts of a module can be fine-tuned (e.g. the vocabulary) and maybe a module does not provides the mechanisms to fine-tune as good as the original (e.g. a way to control dropout or batch norm).

**Comment by saikrishna9494** [comment]
Can you please send me the snippet of code for fine tuning universal sentence vector with a new document.

**Comment by andresusanopinto** [comment]
Have a look in: https://www.tensorflow.org/hub/fine_tuning

**Comment by saikrishna9494** [comment]
Can I create the universal sentence vector model from a different data set other than fine tuning the existing model? Is there any training module available where I can train from my own data from scratch.

**Comment by micahprice** [comment]
There is a great example showing how to transfer learn the universal sentence encoder module while also training a classifier: 
[https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/text_classification_with_tf_hub.ipynb](https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/text_classification_with_tf_hub.ipynb)

But I am wondering how to just train the module itself to output sentence encodings fine tuned to my dataset, and the fine_tuning page referenced above doesn't make it clear how to do that.  My instinct is to train a bogus classifier (real input paragraphs, fake binary classification) and only keep the module, but I would rather do it the correct way.

**Comment by vbardiovskyg** [comment]
Please take a look at https://github.com/tensorflow/hub/issues/46.

**Comment by rsilveira79** [comment]
@vbardiovskyg,  we are testing the Universal Sentence Encoder model, and wanted to double check where was the large-1 model (https://www.tensorflow.org/hub/modules/google/universal-sentence-encoder-large/1) model trained ? I can relate this model to the models from the original paper

**Comment by akhorlin** [comment]
It trained same was as the embedding described in the paper and same way as
https://www.tensorflow.org/hub/modules/google/universal-sentence-encoder/1,
only
this version allows for fine-tuning (trainable=True). This embedding was
published in response to #46 <https://github.com/tensorflow/hub/issues/46>.

**Comment by datistiquo** [comment]
The question was not answered.

is the code für the universal sentence encoder open-sourc? I don't want to finetune rather than train on my owen crawled data as the opener wanted too?

**Comment by bhavsarpratik** [comment]
Has anyone been able to find code for fine-tuning USE transformers?

**Comment by andrew2jara** [comment]
I have the same issue. I do not wish to fine-tune a mode but train an encoder in pytorch
